lib/runtime/router.ts

```diff
@@ -251,6 +251,32 @@ export default class Router extends DenaliObject implements RouterDSL {
   }
 
   /**
+   * Returns the URL for a given action. You can supply a params object which
+   * will be used to fill in the dynamic segements of the action's route (if
+   * any).
+   */
+  urlFor(action: Action | string, data: Object): string {
+    if (typeof action === 'string') {
+      action = this.container.lookup(`action:${ action }`);
+    }
+    if (!action) {
+      return null;
+    }
+    let route: Route;
+
+    forEach(this.routes, (routes) => {
+      route = find(routes, { action: action.constructor });
+      return !route; // kill the iterator if we found the match
+    });
+
+    if (route) {
```

Comment:
This can probably be switched back to `return route ? route.reverse(data) : null` but wasn't sure how to leave it like it was before and make TS happy at the same time.